core/cog-view.c

```diff
@@ -396,3 +442,22 @@ cog_view_get_use_key_bindings(CogView *self)
     CogViewPrivate *priv = cog_view_get_instance_private(COG_VIEW(self));
     return priv->use_key_bindings;
 }
+
+/**
+ * cog_view_get_viewport: (get-property viewport)
+ * @self: A view.
+ *
+ * Gets viewport where the view is attached.
+ *
+ * Returns: The viewport what the view belongs.
+ *
+ * Since: 0.20
+ */
+CogViewport *
+cog_view_get_viewport(CogView *self)
+{
+    g_return_val_if_fail(COG_IS_VIEW(self), NULL);
+
+    CogViewPrivate *priv = cog_view_get_instance_private(self);
+    return priv->viewport;
```

Comment:
No need to change this, but I would have written the same in one line:

```suggestion
    return cog_view_get_instance_private(self)->viewport;
```